pgjdbc/src/main/java/org/postgresql/jdbc/PgSQLInput.java

```diff
@@ -0,0 +1,411 @@
+  }
+
+  @Override
+  public InputStream readBinaryStream() throws SQLException {
+    byte[] bytes = readBytes();
+    return new ByteArrayInputStream(bytes == null ? new byte[0] : bytes);
+  }
+
+  @SuppressWarnings("override.return")
+  @Override
+  public @Nullable Object readObject() throws SQLException {
+    return getNextValue(stringConv);
+  }
+
+  @SuppressWarnings("override.return")
+  @Override
+  public @Nullable <T> T readObject(Class<T> type) throws SQLException {
+    return getNextValue(getConverter(type, connection, timestampUtils));
+  }
+
+  @Override
+  public Ref readRef() throws SQLException {
+    throw Driver.notImplemented(this.getClass(), "readRef()");
+  }
+
+  @Override
+  public Blob readBlob() throws SQLException {
+    throw Driver.notImplemented(this.getClass(), "readBlob()");
+  }
+
+  @Override
+  public Clob readClob() throws SQLException {
+    throw Driver.notImplemented(this.getClass(), "readClob()");
+  }
+
+  @SuppressWarnings("override.return")
+  @Override
+  public @Nullable Array readArray() throws SQLException {
+    return getNextValue(arrayConv);
+  }
+
+  @SuppressWarnings("override.return")
+  @Override
+  public @Nullable URL readURL() throws SQLException {
+    return getNextValue(urlConv);
+  }
+
+  @Override
+  public SQLXML readSQLXML() throws SQLException {
+    return new PgSQLXML(connection, readString());
+  }
+
+  @Override
+  public RowId readRowId() throws SQLException {
+    throw Driver.notImplemented(this.getClass(), "getRowId()");
+  }
+
+  @Override
+  public boolean wasNull() throws SQLException {
+    return wasNull == null ? false : wasNull;
+  }
+
+  @Override
+  public NClob readNClob() throws SQLException {
+    throw Driver.notImplemented(this.getClass(), "readNClob()");
+  }
+
+  @Override
+  public String readNString() throws SQLException {
+    throw Driver.notImplemented(this.getClass(), "readNString()");
+  }
+
+  //
+  // NOTE: Without the @SuppressWarnings("nullness")
+  //   error: [argument] incompatible argument for parameter arg0 of Array.set.
+  //       java.lang.reflect.Array.set(results, i, converter.apply(item));
+  //                                   ^
+  //   found   : @Initialized @NonNull Object @Initialized @Nullable []
+  //   required: @Initialized @NonNull Object
+  //
+  @SuppressWarnings("nullness")
+  private static Object reflectArray(Class<?> itemType, SQLFunction<String, ?> converter, String value) throws SQLException {
+    List<@Nullable String> items = new SQLDataReader().parseArray(value);
+    Object @Nullable [] results = (Object @Nullable []) java.lang.reflect.Array.newInstance(itemType, items.size());
+    for (int i = 0; i < items.size(); i++) {
+      @Nullable String item = items.get(i);
+      if ("NULL".equals(item)) {
+        java.lang.reflect.Array.set(results, i, null);
+      } else {
+        java.lang.reflect.Array.set(results, i, converter.apply(item));
+      }
+    }
+    return results;
+  }
+
+  // //
```

Comment:
is there a reason to keep this here ?

Reply:
No, was just keeping until this was all resolved. Cheers.